portal-bridge/src/bridge/state.rs

```diff
@@ -0,0 +1,226 @@
+                    .expect("Getting state proof shouldn't fail block: {block_index}");
+                let content_key = create_content_key(&account_proof)
+                    .expect("We should bew able to create state content key block: {block_index}");
+                let content_value =
+                    create_content_value(block_tuple.header.header.hash(), &account_proof).expect(
+                        "We should bew able to create state content value block: {block_index}",
+                    );
+                Self::spawn_serve_state_proof(
+                    self.portal_clients.clone(),
+                    content_key,
+                    content_value,
+                    permit,
+                    self.metrics.clone(),
+                );
+            }
+        }
+    }
+
+    fn spawn_serve_state_proof(
+        portal_clients: Vec<HttpClient>,
+        content_key: StateContentKey,
+        content_value: StateContentValue,
+        permit: OwnedSemaphorePermit,
+        metrics: BridgeMetricsReporter,
+    ) -> JoinHandle<()> {
+        info!("Spawning serve_state_proof for content key: {content_key}",);
+        tokio::spawn(async move {
+            let timer = metrics.start_process_timer("spawn_serve_state_proof");
+            match timeout(
+                SERVE_BLOCK_TIMEOUT,
+                Self::serve_state_proof(
+                    portal_clients,
+                    content_key.clone(),
+                    content_value,
+                    metrics.clone()
+                )).await
+                {
+                Ok(result) => match result {
+                    Ok(_) => {
+                        debug!("Done serving state proof: {content_key}");
+                    }
+                    Err(msg) => warn!("Error serving state proof: {content_key}: {msg:?}"),
+                },
+                Err(_) => error!("serve_state_proof() timed out on state proof {content_key}: this is an indication a bug is present")
+            };
+            drop(permit);
+            metrics.stop_process_timer(timer);
+        })
+    }
+
+    async fn serve_state_proof(
+        portal_clients: Vec<HttpClient>,
+        content_key: StateContentKey,
+        content_value: StateContentValue,
+        metrics: BridgeMetricsReporter,
+    ) -> anyhow::Result<()> {
+        let timer = metrics.start_process_timer("gossip_state_content");
+        match gossip_state_content(&portal_clients, content_key.clone(), content_value).await {
+            Ok(_) => {
+                debug!("Successfully state proof: {content_key}")
+            }
+            Err(msg) => {
+                warn!("Error state proof: {content_key} {msg}",);
+            }
+        }
+        metrics.stop_process_timer(timer);
+        Ok(())
+    }
+}
+
+/// Fetches era1 files hosted on era1.ethportal.net and shuffles them
+async fn get_list_of_era1_files(http_client: &Client) -> anyhow::Result<Vec<String>> {
```

Comment:
Can you use [this util](https://github.com/ethereum/trin/blob/master/portal-bridge/src/bridge/era1.rs#L462)?